docs/designs/config.md

```diff
@@ -0,0 +1,290 @@
+metadata:
+  breadcrumb_path: ~/breadcrumb/toc.yml
+  searchScope: ["sql"]
+contributors:
+  exclude:
+  - hexiaokai
+  - openpublishingbuild
+  - sudeepku
+  - saldana
+  - iRaindrop
+monikerDefinition: https://api.docs.com/monikers/
+```
+No config is set to group level.
+* `monikerRange`: Use *file level config* pattern.
+
+### `MSGraph-Reference`: Markdown Fragments
+[docfx.json](https://github.com/MicrosoftDocs/MSGraph-Reference/blob/master/docfx.json)
+```yml
+name: MSGraphDocs_Reference
+basePath: rest
+product: MSDN
+content: docs-ref-autogen/**
+routes:
+  docs-ref-autogen/1.0: /rest/api
+  docs-ref-autogen/1.0/toc.yml: /rest/api/toc/toc.json
+  docs-ref-autogen/beta: /rest/api
+  docs-ref-autogen/beta/toc.yml: /rest/api/toc/toc.json
+fragments:
+  docs-ref-autogen: docs-ref-authored
+metadata:
+  values:
+    breadcrumb_path: /rest/breadcrumb/toc.json
+    extendBreadcrumb: true
+monikerRanges:
+  docs-ref-autogen/1.0/: graph-rest-1.0
+  docs-ref-autogen/beta/: graph-rest-beta
+monikerDefinition: https://api.docs.com/monikers/
+```
+* `fragments`: an object specify the fragments folder of the contents
+* `routes`: specify the published URL. Note that the key is not glob pattern, it's a **folder** or a **file**. Glob pattern is ambiguous to express which part of path is not needed in publish URL.
+
+## Others
+Some configs not needed in *Phase 1* are not discussed in this spec. They will not introduce breaking changes in current design. Like:
+1. Configs for other features, like: `validation`.
+2. Configs for other programming language CI tools, like the `metadata` command in v2.
+
+## Open Questions:
+### 1. Zero downtime basePath change support
+After abandoning `docset`, writers can use relative path to link cross docsets (means contents with different `name` in v3). If the basePath of one docset changes in DHS, the published URL changes immediately before next publish. Before next publish, the relative path is wrong.
+
+To fix this, there is two solutions:
+1. When resolving links, DocFX checks whether the target and source has the same `name`. If not, use **absolute path** for resolve link.
+2. When changing basePath in DHS, the published URL will not be updated until the next build is finish. Actually, after the config change in DHS, the basePath in `docfx.yml` should also be synced immediately, which will trigger a build.
+
+As #1 will introduce complexity when resolve link, #2 is recommended here. It also benefits data consistency for other similar config changes.
+
+### 2. How to support branch mapping?
+It is not suposed to be supported for now. If it is supported, there is 2 options:
+1. Add branch_mapping to gitDependencies.
+2. Generalize the concept to branch-aware config.
+
+### 3. Can we restrict one `name` for one `repo`?
+If so, `name` needn't be an object. That's to say, all "docsets" in a repo share the same name.
+
+### 4. DocumentId might need a dedicated section
+A lot of configs is related to it.
```

Comment:
have we considered the migration of current `docfx.json` to `docfx.yml`? will there be a tool to help with the migration? how to move OPS to consume `docfx.yml` in short term?

Reply:
Will discuss with Feng about the details.

---
In reply to: [183949261](https://github.com/dotnet/docfx/pull/2644#discussion_r183949261) [](ancestors = 183949261)